devito/data/data.py

```diff
@@ -260,6 +388,176 @@ def _convert_index(self, glb_idx):
 
         return loc_idx[0] if len(loc_idx) == 1 else tuple(loc_idx)
 
+    def _set_global_idx(self, val, idx, val_idx):
```

Comment:
this could be a free function (self isn't actually used).

Reply:
also, I'm not sure I like it be called "_set" which makes one think it's changing internal state?

Reply:
reminder

Reply:
reminder. We should really drop the _set

Reply:
also: I **think** ``val_idx`` should actually be ``glb_idx`` ?